client/src/pages/List.js

```diff
@@ -1,20 +1,24 @@
-import React, { useState } from "react";
-import { useParams } from "react-router-dom";
+import React, { useEffect, useState } from "react";
+import { Link, useParams } from "react-router-dom";
 import { fetchList } from "../api/list";
 import styled from "@emotion/styled";
 import { postProduct, fetchProductByname } from "../api/products";
 import useAsync from "../hooks/useAsync";
 import Header from "../components/Header";
+import PropTypes from "prop-types";
 
-function List() {
+function List({ setListData }) {
   const { id } = useParams();
   const [query, setQuery] = useState([]);
   const [display, setDisplay] = useState(false);
   const [products, setProducts] = useState([]);
   const { data: list, loading, error, refetch } = useAsync(fetchList, id);
 
+  useEffect(() => {
+    setListData(list);
+  }, [setListData, list]);
```

Comment:
do you need setListData as dependency? this would not change, wouldn't it? looks odd ;-)

Reply:
Eslint is marking this as missing dependency, so I included it. Otherwise its running a problem :/

Reply:
my eslinter is barking at this moment to either include or to not set any at all. So, to avoid displaying a problem I added it